sos/archive.py

```diff
@@ -209,7 +210,14 @@ def add_node(self, path, mode, device):
         dest = self.dest_path(path)
         self._check_path(dest)
         if not os.path.exists(dest):
-            os.mknod(dest, mode, device)
+            try:
+                os.mknod(dest, mode, device)
+            except OSError as e:
```

Comment:
You'd obviously either need to do this in your own mount namespace (which the kernel will flush when you exit) or keep a list of those bind-mounts and unmount them on exit.